### PR TITLE
Migrate the certs.sh e2e test to ginkgo

### PR DESCRIPTION
**Pull request by a-robinson**
Fixes issue #4185.

However, this is somewhat hacky. We need to SSH to check that the files on the master. gcloud ssh is the most reasonable way to ssh to a GCE VM, but it requires the project, zone, and VM name, which aren't currently available from within our ginkgo tests. This gets rid of one of the few remaining bash tests, but is more complicated than the original bash test.

cc @zmerlynn

### CODE REVIEW

**Comment by zmerlynn** [comment]
cc @filbranden  ... thoughts? I'm trying to think of how to make this less horrible.

**Comment by filbranden** [comment]
Given the gcloud interface, I really don't see a lot on how to improve this...

I think @satnam6502 would request that you use command line flags instead of environment variables, but I'm not 100% sure here it would actually improve things.

**Comment by zmerlynn** [comment]
I feel like command line flags would actually clutter things more. :(

**Comment by a-robinson** [comment]
Using command line flags would definitely clutter things up, and also make the GCE/GKE specific bits more visible to the tests that aren't supposed to be GCE/GKE specific.

**Comment by filbranden** [comment]
Big part of the problem with relying on environment variables is that it makes it hard to run `cmd/e2e` as a standalone binary. Though it's probable that that's already not really possible today...

I think a good approach would be to have some kind of YAML/JSON configuration file for the e2e cluster and that should be all the configuration you need to run it. `e2e.sh -up` would create such a file and the other steps would consume it. Does that sound reasonable?

**Comment by satnam6502** [comment]
I find it hugely valuable to run e2e tests from the command line and select just a specific test to run and know for sure how all the environmental information is set by looking at the flags. If tests depend on environment variables I find it hard to run tests in such a manner and I am often confused about how the environment is set. Clutter? Perhaps. But I think being explicit is valuable.

**Comment by a-robinson** [comment]
Ok, I'll see what moving this into flags or a config file looks like.

**Comment by a-robinson** [comment]
Updated to use flags for the GCE config info. How does this look comparatively?

File: hack/ginkgo-e2e.sh [hunk 1]
```diff
@@ -102,5 +102,8 @@ fi
 "${e2e}" "${auth_config[@]:+${auth_config[@]}}" \
   --host="https://${KUBE_MASTER_IP-}" \
   --provider="${KUBERNETES_PROVIDER}" \
+  --gce_project="${PROJECT}" \
+  --gce_zone="${ZONE}" \
```
Comment (zmerlynn) [inline hack/ginkgo-e2e.sh]: Won't these trip `nounset`? Defaulting them to empty should be fine, though.

**Comment by zmerlynn** [comment]
This LGTM. I would really rather us move to a yaml file at some point, but I'm not going to ask you to do that here.

File: hack/ginkgo-e2e.sh (hunk 1 above)
Comment (a-robinson) [inline hack/ginkgo-e2e.sh]: Good call, fixed by defaulting to empty above.

**Comment by a-robinson** [comment]
ping - any final opinions, or should I go ahead and merge?

**Comment by zmerlynn** [comment]
LGTM.

**Comment by zmerlynn** [comment]
I'm not sure how, but it looks like this broke auth LoadClient. Jenkins was very confused. Reverting for the night.

**Comment by zmerlynn** [comment]
Sample error:
```
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/rc.go:41
Expected error:
    <*errors.errorString | 0xc2080c1d10>: {
        s: "Error loading auth: stat /var/lib/jenkins/jobs/kubernetes-e2e-gce/workspace/.kube/_e2e-test-jenkins/kubernetes_auth: no such file or directory",
    }
    Error loading auth: stat /var/lib/jenkins/jobs/kubernetes-e2e-gce/workspace/.kube/_e2e-test-jenkins/kubernetes_auth: no such file or directory
not to have occurred
```

**Comment by satnam6502** [comment]
See also my woes at #4642

**Comment by satnam6502** [comment]
I'm in good shape after the revert of #4644

**Pull request merged**